Automattic/wp-calypso
# Gutenlypso: Update edit-post to the latest version

#### Changes proposed in this Pull Request

* Update the `edit-post` folder with the latest and greatest [Core `edit-post` package](https://github.com/WordPress/gutenberg/tree/master/packages/edit-post/src/).

All the changes and additions to the `edit-post` folder are marked with a `GUTENLYPSO` comment.
The deletions are unfortunately not marked (my bad, I noticed it way too late).

#### Freebies

The block navigation:
![screenshot 2018-12-04 at 17 58 15](https://user-images.githubusercontent.com/2070010/49462740-b07c1880-f7ee-11e8-930f-aad9837cb57f.png)

The more menu (now 99% identical to Core, and moved back into the `edit-post` folder):
![screenshot 2018-12-04 at 17 58 31](https://user-images.githubusercontent.com/2070010/49462762-bf62cb00-f7ee-11e8-8497-e0afce32ae88.png)

The options modal (without the "enable NUX tips" options):
![screenshot 2018-12-04 at 17 58 36](https://user-images.githubusercontent.com/2070010/49462802-e02b2080-f7ee-11e8-83d1-49beb1a66ca5.png)

The keyboard shortcuts modal (all of them _should_ be working; some that weren't working before are working now though):
![screenshot 2018-12-04 at 17 58 50](https://user-images.githubusercontent.com/2070010/49462890-1e284480-f7ef-11e8-86fd-c4a3927ba80f.png)

The preview now has the nice "generating" animation 💖:
![screenshot 2018-12-04 at 17 59 51](https://user-images.githubusercontent.com/2070010/49462912-284a4300-f7ef-11e8-8ee8-c921815a1b1e.png)

#### Testing instructions

Make sure there are no obvious regressions on `/gutenberg`.

Pay special attention to:
- How the header looks.
- The preview and publish flows.
- The more menu (e.g. the Keyboard Shortcuts and Options modals).
- The sidebar.


Fixes #28989 #28990
